.cloudbuild/prombench.yaml

```diff
@@ -0,0 +1,335 @@
+# https://cloud.google.com/build/docs/build-config-file-schema#options
+options:
+  env:
+  - 'CANCEL_BENCHMARK=false'
+  - 'CLUSTER_NAME=prombench'
+  - 'CREATE_CLUSTER=true'
+  - 'DASHBOARD_ID=$_DASHBOARD_ID'
+  - 'DOMAIN_NAME='
+  - 'GITHUB_STATUS_TARGET_URL='
+  - 'GKE_PROJECT_ID=$_GKE_PROJECT_ID'
+  - 'GRAFANA_ADMIN_PASSWORD=admin'
+  - 'LAST_COMMIT_SHA='
+  - 'OAUTH_TOKEN='
+  - 'PROMBENCH_IMAGE=$_PROMBENCH_IMAGE'
+  - 'PROMETHEUS_IMAGE_REPOSITORY=gke.gcr.io/prometheus-engine/prometheus'
+  - 'PROVIDER=gke'
+  - 'SERVICEACCOUNT_CLIENT_EMAIL=$_SERVICEACCOUNT_CLIENT_EMAIL'
+  - 'WH_SECRET='
+  - 'ZONE=us-east4-a'
+  substitutionOption: ALLOW_LOOSE
+  dynamicSubstitutions: true
+
+# https://cloud.google.com/build/docs/configuring-builds/substitute-variable-values
+substitutions:
+  _COMMENTER: $(body.comment.user.login)
+  _COMMENTER_ROLE: $(body.comment.author_association) # OWNER / MEMBER / etc. (https://docs.github.com/en/graphql/reference/enums#commentauthorassociation)
+  _GIT_REPO: $(body.repository.full_name) # GoogleCloudPlatform/prometheus
+  _PR_COMMENT_ACTION: $(body.action)
+  _PR_COMMENT_TEXT: $(body.comment.body)
+  _PR_NUMBER: $(body.issue.number)
+
+# https://cloud.google.com/build/docs/build-config-file-schema#availablesecrets
+availableSecrets:
+  secretManager:
+  - versionName: projects/${PROJECT_NUMBER}/secrets/prombench_service_account_adc_json/versions/1
+    env: 'ADC'
+  - versionName: projects/${PROJECT_NUMBER}/secrets/bot-github-personal-access-token/versions/1
+    env: 'GH_PAT'
+
+
+
+steps:
+  - id: "Verifying GitHub comment and user permissions"
+    name: bash
+    env:
+    - '_COMMENTER=$_COMMENTER'
+    - '_COMMENTER_ROLE=$_COMMENTER_ROLE'
+    - '_GIT_REPO=$_GIT_REPO'
+    - '_PR_COMMENT_ACTION=$_PR_COMMENT_ACTION'
+    - '_PR_COMMENT_TEXT=$_PR_COMMENT_TEXT'
+    - '_PR_NUMBER=$_PR_NUMBER'
+    secretEnv: ['GH_PAT']
+    script: |
+      #!/usr/bin/env bash
+
+      # This function posts comments to GitHub PRs
+      function post_to_github() {
+        URL="$1"
+        BODY="$2"
+
+        curl --silent -L -X POST \
+          -H "Accept: application/vnd.github+json" \
+          -H "Authorization: Bearer $${GH_PAT}" \
+          -H "X-GitHub-Api-Version: 2022-11-28" \
+          "$${URL}" \
+          -d '{"body": "'"$${BODY}"'"}'
+      }
+
+      # Checking the action on the PR (benchmark should trigger only when a new comment is created)
```

Comment:
Having multiple checks instead of combining them into a single `if` gives us granularity while logging. We can check the logs to understand why something might have gone wrong if something goes wrong.
Combining everything in a single `if` means that even if the error message gets logged, it still needs to be decoded to understand why something went wrong (which exact requirement was not met).